db/document.go

```diff
@@ -491,24 +494,32 @@ func parseXattrStreamData(xattrName string, data []byte) (body []byte, xattr []b
 	for pos < xattrsLen {
 		pairLen := binary.BigEndian.Uint32(data[pos : pos+4])
 		if pairLen == 0 || int(pos+pairLen) > len(data) {
-			return nil, nil, fmt.Errorf("Unexpected xattr pair length (%d) - unable to parse xattrs", pairLen)
+			return nil, nil, nil, fmt.Errorf("Unexpected xattr pair length (%d) - unable to parse xattrs", pairLen)
 		}
 		pos += 4
 		pairBytes := data[pos : pos+pairLen]
 		components := bytes.Split(pairBytes, separator)
 		// xattr pair has the format [key]0x00[value]0x00, and so should split into three components
 		if len(components) != 3 {
-			return nil, nil, fmt.Errorf("Unexpected number of components found in xattr pair: %s", pairBytes)
+			return nil, nil, nil, fmt.Errorf("Unexpected number of components found in xattr pair: %s", pairBytes)
 		}
 		xattrKey := string(components[0])
 		// If this is the xattr we're looking for , we're done
 		if xattrName == xattrKey {
-			return body, components[1], nil
+			xattr = components[1]
+		}
+		if userXattrName != "" && userXattrName == xattrKey {
+			userXattr = components[1]
+		}
```

Comment:
Nitpick (only if other changes): Won't get into the 2nd if anyway, but I think it makes it clearer to the reader that it's only one or the other.

```suggestion
			xattr = components[1]
		} else if userXattrName != "" && userXattrName == xattrKey {
			userXattr = components[1]
		}
```